index.ts

```diff
@@ -260,21 +260,20 @@ export async function getLinkPreview(
     throw new Error(`link-preview-js did not receive a valid url or text`);
   }
 
-  const detectedUrl = (options?.proxyUrl ? options.proxyUrl.concat(text) : text)
-    .replace(/\n/g, ` `)
-    .split(` `)
-    .find((token) => CONSTANTS.REGEX_VALID_URL.test(token));
+  const detectedUrl = text.replace(/\n/g, ` `).split(` `).find((token) => CONSTANTS.REGEX_VALID_URL.test(token));
 
   if (!detectedUrl) {
     throw new Error(`link-preview-js did not receive a valid a url or text`);
   }
 
   const fetchOptions = { headers: options?.headers ?? {} };
 
+  const fetchUrl = options?.proxyUrl ? options.proxyUrl.concat(detectedUrl) : detectedUrl;
+
   try {
-    const response = await fetch(detectedUrl, fetchOptions);
+    const response = await fetch(fetchUrl, fetchOptions);
```

Comment:
looks better! now the only thing missing is the test 👍

Reply:
added a test